compiler/sem/semtypes.nim

```diff
@@ -56,7 +56,7 @@ proc semEnum(c: PContext, n: PNode, prev: PType): PType =
       if n[i][0].kind == nkPragmaExpr:
         e = newSymS(skEnumField, n[i][0][0], c)
         identToReplace = addr n[i][0][0]
-        n[i][0][1] = pragma(c, e, n[i][0][1], enumFieldPragmas)
+        n[i][0][1] = pragmaDecl(c, e, n[i][0][1], enumFieldPragmas)
         # check if we got any errors and if so report them
         for e in ifErrorWalkErrors(c.config, n[i][0][1]):
```

Comment:
I reallly like the direction of the PR. I think once these changes are in, I can swap out these walk errors with propagating the errors instead. That should clean-up a bunch of downstream code. 🎉